lib/travis/build/script/node_js.rb

```diff
@@ -169,6 +194,12 @@ def setup_npm_cache
           def iojs_3_plus?
             (config[:node_js] || '').to_s.split('.')[0].to_i >= 3
           end
+
+          def install_yarn
+            sh.cmd    "curl -o- -L https://yarnpkg.com/install.sh | bash", echo: true
```

Comment:
I wonder if it's worth adding an option to specify which version of Yarn to use? We have a GitHub issue to add a `--version` flag to the install script (see https://github.com/yarnpkg/yarn/issues/1856) but you could achieve it today by downloading the right tarball (eg. https://yarnpkg.com/downloads/0.16.1/yarn-v0.16.1.tar.gz) and extracting it yourself. That's essentially what the install script is doing 😄